coremis_app_integration/endpoints/login.py

```diff
@@ -0,0 +1,59 @@
+from django.utils.translation import gettext_lazy as _
+from rest_framework.authentication import authenticate
+from rest_framework.decorators import (
+    authentication_classes,
+    permission_classes,
+    api_view
+)
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework import status
+
+from coremis_app_integration.authorization import (
+    CustomBasicAuthentication,
+    DETAILS,
+    FAILED,
+    SUCCESS
+)
+from payroll.models import PaymentPoint
+
+
+@api_view(['POST'])
+@authentication_classes([CustomBasicAuthentication])
+@permission_classes([IsAuthenticated])
+def login(request):
+    # check if user is assigned to any PaymentPoint entity which indicates ppm user role
+    username = request.data.get('username', None)
+    password = request.data.get('password', None)
+    authenticated_user = authenticate(username=username, password=password)
+    if authenticated_user:
+        payment_points = PaymentPoint.objects.filter(ppm__username=username)
+        if payment_points.count() > 0:
+            response_data = SUCCESS
+            response_data["result"] = _build_success_output(authenticated_user)
+            return Response(response_data, status=status.HTTP_200_OK)
```

Comment:
i see several repetition of action like: 
x = dict 
x[key] = sth 
return Result(x, status) 
it'd be better create some response factory directly, or change SUCCESS to dataclass and use constructor to build the response body. 

Also, I think that assigning dict is not safe for the same reason it should be avoided in adding dict as a function parameter (it's not a new instance).

Reply:
changed to dataclass